src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -97,6 +91,12 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
                 return;
             }
 
+            var expressionSymbol = semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol;
+            if (expressionSymbol is IParameterSymbol && !syntaxFacts.IsLocalFunctionStatement(containingMethod))
+            {
+                return;
+            }
```

Comment:
i think the original fix was corrrect.  i don't see why we'd want to special case local functions.

note: we should have extra tests here.  for example if the user is on a local-func parameter, we shouldn't add that a second time either.

Reply:
IMO the correct fix is none of the two, but:

```csharp
            if (expressionSymbol is IParameterSymbol parameterSymbol && parameterSymbol.ContainingSymbol.Equals(containingSymbol))
            {
                return;
            }
```

Reply:
i'm ok not adding one parameter to another container.  but i don't feel strongly about that.

Reply:
@CyrusNajmabadi This scenario is important IMO if I want to move from:

```csharp
public class C
{
    public void M(string s)
    {
        localFunction();

        void localFunction()
        {
            _ = s.ToString();
        }
    }
}
```

to:

```csharp
public class C
{
    public void M(string s)
    {
        localFunction(s);

        localFunction("Another string");

        void localFunction(string s)
        {
            _ = s.ToString();
        }
    }
}
```

Reply:
@Youssef1313 I agree, I like that example.